pkgs/development/compilers/llvm/4/clang/default.nix

```diff
@@ -13,20 +13,32 @@ let
       mv clang-tools-extra-* $sourceRoot/tools/extra
     '';
 
-    buildInputs = [ cmake libedit libxml2 llvm python ];
+    buildInputs = [ cmake libedit libxml2 llvm python python.pkgs.sphinx ];
 
     cmakeFlags = [
       "-DCMAKE_CXX_FLAGS=-std=c++11"
+      "-DCLANG_INCLUDE_DOCS=ON"
+      "-DLLVM_ENABLE_SPHINX=ON"
+      "-DSPHINX_OUTPUT_MAN=ON"
+      "-DSPHINX_OUTPUT_HTML=OFF"
+      "-DSPHINX_WARNINGS_AS_ERRORS=OFF"
     ] ++
     # Maybe with compiler-rt this won't be needed?
     (stdenv.lib.optional stdenv.isLinux "-DGCC_INSTALL_PREFIX=${gcc}") ++
     (stdenv.lib.optional (stdenv.cc.libc != null) "-DC_INCLUDE_DIRS=${stdenv.cc.libc}/include");
 
     patches = [ ./purity.patch ];
 
+    postBuild = ''
+      cmake --build . --target docs-clang-man
+    '';
+
     postPatch = ''
       sed -i -e 's/Args.hasArg(options::OPT_nostdlibinc)/true/' lib/Driver/Tools.cpp
       sed -i -e 's/DriverArgs.hasArg(options::OPT_nostdlibinc)/true/' lib/Driver/ToolChains.cpp
+
+      # Patch for standalone doc building
+      sed -i '1s,^,find_package(Sphinx REQUIRED)\n,' docs/CMakeLists.txt
     '';
 
     outputs = [ "out" "python" ];
```

Comment:
Hmm, is there a good way to avoid including the manpages in stdenv but still include them when installing the compiler explicitly?

Looking into it, looks like [`cc-wrapper` propagates the `man` output](https://github.com/NixOS/nixpkgs/blob/7c8a060c09799eb2ee70c00aa695ff08e5f07c6f/pkgs/build-support/cc-wrapper/default.nix#L160) which I believe satisfies the "make available when explicitly installed" part of my question above.

Neat re:installing man pages globally without their binaries, I hadn't thought about that and it seems very useful indeed!

I'll make that change shortly, although I don't think i can check whether the Darwin stdenv ends up containing the man page or not.